craft_providers/multipass/multipass_provider.py

```diff
@@ -0,0 +1,154 @@
+    try:
+        subprocess.run(["brew", "cask", "install", "multipass"], check=True)
+    except subprocess.CalledProcessError as error:
+        raise errors.MultipassInstallationError(
+            "error during brew installation",
+            details=details_from_called_process_error(error),
+        ) from error
+
+
+def _install_linux() -> None:
+    try:
+        subprocess.run(["sudo", "snap", "install", "multipass"], check=True)
+    except subprocess.CalledProcessError as error:
+        raise errors.MultipassInstallationError(
+            "error during snap installation",
+            details=details_from_called_process_error(error),
+        ) from error
+
+
+def _install_windows() -> None:
+    raise errors.MultipassInstallationError(
+        "automated installation not yet supported for Windows"
+    )
+
+
+class MultipassProvider:  # pylint: disable=no-self-use
+    """Multipass Provider."""
+
+    def _configure_instance(
+        self,
+        *,
+        instance: MultipassInstance,
+        base_configuration: Base,
+    ) -> None:
+        """Configure instance.
+
+        Start to instance first to ensure it is started, as well as to cancel
+        any outstanding delay-shtudown request.
+
+        :param instance: Instance to configure.
+        :param base_configuration: Base configuration to apply.
+        """
+        instance.start()
+
+        base_configuration.setup(executor=instance)
+
+    def create_instance(
+        self,
+        *,
+        base_configuration: Base,
+        cpus: int = 2,
+        disk_gb: int = 256,
+        image_name: str,
+        mem_gb: int = 2,
+        name: str,
+    ) -> MultipassInstance:
+        """Create, start, and configure instance.
+
+        :param name: Name of instance.
+        :param base_configuration: Base configuration to apply to instance.
+        :param cpus: Number of CPUs.
+        :param disk_gb: Disk allocation in gigabytes.
+        :param image_name: Multipass image to use, e.g. snapcraft:core20.
+        :param mem_gb: Memory allocation in gigabytes.
+
+        :returns: Multipass instance.
+        """
+        multipass = Multipass()
+        instance = MultipassInstance(
+            name=name,
+            multipass=multipass,
+        )
+
+        if instance.exists():
+            self._configure_instance(
+                instance=instance,
+                base_configuration=base_configuration,
+            )
+
+        # Re-check if instance exists as it may been cleaned.
+        # If it doesn't exist, launch a fresh instance.
+        if not instance.exists():
+            instance.launch(
+                cpus=cpus,
+                disk_gb=disk_gb,
+                mem_gb=mem_gb,
+                image=image_name,
+            )
+
+        self._configure_instance(
+            instance=instance,
+            base_configuration=base_configuration,
+        )
+        return instance
+
+    def install(self) -> str:
```

Comment:
souldn't this be a static class method?

Reply:
Probably should just drop the class altogether.